README.md

```diff
@@ -1544,6 +1544,28 @@ result := lo.Ternary[string](false, "a", "b")
 // "b"
 ```
 
+### TernaryF
+
+```go
+result := lo.TernaryF[string](true, func() string { return "a" }, func() string { return "b" })
+// "a"
+
+result := lo.TernaryF[string](false, func() string { return "a" }, func() string { return "b" })
+// "b"
+```
+
+Useful to avoid nil-pointer dereferencing in intializations
+
+```go
+type S struct {
+    Val *string
+}
+
+s := S{}
+
+someStr := TernaryF[string](s.Val == nil, func() string { return "defaultValue" }, func() string { return *s.Val })
+```
```

Comment:
It would also be useful to avoid running code for no reason.

One suggestion:
I think you can make the example more concise and demonstrate that no useless code will be run like that:
```go
var s *string
someStr := TernaryF[string](s == nil, func() string { return uuid.New().String() }, func() string { return *s })
```

WDYT?

Reply:
That works too. I just made the example a reference to the problem I was facing that lead to the PR.  I'll make the update here shortly